requests_kerberos/kerberos_.py

```diff
@@ -71,6 +90,9 @@ def _negotiate_value(response):
         regex = re.compile('(?:.*,)*\s*Negotiate\s*([^,]*),?', re.I)
         _negotiate_value.regex = regex
 
+    if response is None: # we're doing an initial request- just use an empty negotiate value
+        return ''
+
```

Comment:
That's similar to the way I originally did it, but when doing stuff like this (for a project I don't "own") I tend to go for minimal code change (for ease of review) rather than making bigger changes to the execution flow. I don't have strong feelings either way, but you make a reasonable argument that I can go back to the original way.